Update ehighways resolution shape generation

brynpickering opened the pull request:
Supercedes #336
Fixes #318 

This is in response to a question you had @timtroendle around defining the shape source in `shapes` in the config. I went to clean up the unresolved comments from #336 and realised that it is a lot cleaner to take that approach. This PR leads to a less flexible solution - E-Highways is baked in more than in #336 - but it is cleaner.

As with #336, it only works with NUTS2013, but I will open an issue about that and have a fix ready to go that I'll add via another PR.
 
## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [x] Tests added to cover contribution
- [x] Documentation updated
- [x] Configuration schema updated

timtroendle commented on scripts/shapes/units.py:
```diff
@@ -3,95 +3,145 @@
-    return pycountry.countries.lookup(country_name).alpha_3
+def _continental_layer(units: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    "For the continental resolution, dissolve all"
+    #
```
Remove

timtroendle commented on scripts/shapes/units.py:
```diff
@@ -3,95 +3,145 @@
 
-def _iso3(country_name):
-    return pycountry.countries.lookup(country_name).alpha_3
+def _continental_layer(units: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
```
Consider renaming to `merge_to_continental_layer` or similar.

timtroendle reviewed (approved):
I am liking this a lot. Just a few minor comments.

timtroendle reviewed (changes requested):
CHANGELOG and documentation updates are missing.

brynpickering commented on config/schema.yaml:
```diff
@@ -517,6 +521,7 @@ properties:
                                 - "Norway"
                                 - "Serbia"
                                 - "Switzerland"
+                                - "Iceland" # added for ehighways
```
sorry, slipped through the net

brynpickering commented on envs/default.yaml:
```diff
@@ -16,7 +16,7 @@ dependencies:
     - hdf5=1.10
     - h5py=3.1.0
-    - libnetcdf=4.7
+    - libnetcdf=4.8.0
```
I don't have a linux device to try it on, so we'll have to wait for one of the others

brynpickering commented:
@timtroendle I've addressed your issues but...

On running the minimal workflow with updated zenodo pointer (to new potentials data that includes `ehighways`), a previous issue rears its head (#99). This would be solved by merging in #239.

timtroendle commented:
Will have a look at #239 soon.

brynpickering commented:
Minimal workflow now runs successfully, including `all_tests` (which covers the `ehighways` resolution as well as the `regional` one).

timtroendle commented on config/schema.yaml:
```diff
@@ -396,6 +396,10 @@ properties:
+            nuts-year:
+                description: Version of the Nomenclature of Territorial Units for Statistics to use in creating spatial units (see https://ec.europa.eu/eurostat/web/nuts/history for more info).
+                type: number
+                enum: [2013]  # FIXME: enable other years
```
Doesn't sound like a fixme to me.
